enhancements/cluster-logging/cluster-logging-es-rollover-data-design.md

```diff
@@ -0,0 +1,253 @@
+ 
+
+### Non-Goals
+
+This change will not:
+* Provide a general data management policy API that fully exposes Elasticsearch's rollover API
+
+## Proposal
+
+This proposal introduces two specific changes to to achieve its goals:
+
+* Co-located data in a few opinionated indices
+* Index management using [rollover index API](https://www.elastic.co/guide/en/elasticsearch/reference/6.3/indices-rollover-index.html)
+
+Logs of a given type (e.g. app container, infra container, node) are separated by index.  The Cluster Logging collector writes logs to a well-known alias established by the `cluster-logging-operator`.  The ClusterLogging CR instance specifies the management policy for each log type index.  This controls: 
+* maximum number of logs per index (e.g. 30G) 
+* the maximum age (e.g. 7 days).  
+
+This policy is passed to the Elasticsearch CR for the `elasticsearch-operator` to manage the rollover policy.
+
+
+### Implementation Details
+
+#### Assumptions
+
+* ClusterLogging will expose the minimal needed set of Elasticsearch rollover policy management API in order to achieve the previously described goals
+* ClusterLogging will manage rollover as Elasticsearch index management is either restricted by Elastic licensing or not available in the opensource version of OpenDistro
+* Security will be addressed by using the OpenDistro security plugin and document level security (DLS). Details TBD.
+
+#### Data Model
+Logs of a given type are co-located to the following indices:
+
+| Log Type | Read Alias | Write Alias | Initial Index |
+|---------|-----------|---------------|---------------|
+| Node (`logs.infra`)|infra, infra.journal|infra.journal-write|infra.journal-00001|
+| Container Infra (`logs.infra`)|infra, infra.container|infra.container-write|infra.container-000001|
+| Container Application (`logs.app`)|app.logs|app.container-write|app.container-000001|
+
+**Note:** Log types are further defined in [LogForwarding](./cluster-logging-log-forwarding.md).
+
+#### ClusterLogging API
+```
+apiVersion: "logging.openshift.io/v1"
+kind: "ClusterLogging"
+metadata:
+  name: "instance"
+spec:
+  logStore:
+    retentionPolicy: 
+      logs.app:
+        maxAge: 7d
+      logs.infra
+        maxAge: 7d
+
+```
+#### Cluster Logging Operator
+The `cluster-logging-operator` will utilized the ClusterLogging CR retention policy to spec
+the desired aforementioned Elasticsearch CR indexManagement.
+#### Elasticsearch API
+```
+apiVersion: "logging.openshift.io/v1"
+kind: "Elasticsearch"
+metadata:
+  name: "elasticsearch"
+spec:
+  indexManagement:
+    policies:
+    - name: infra-policy
+      phases:
+        hot:
+          actions:
+            rollover:
+              max_size:  30GB         #opinionated value on Elastic recommendations
+              max_age:   3d
+        delete:
+          min_age: 7d
+    templates:
```

Comment:
i think we should abstract this slightly... this is a lot of knobs. maybe we can tuck this under something similar to what you have for `retentionPolicy` in the ClusterLogging api

Reply:
> i think we should abstract this slightly... this is a lot of knobs. maybe we can tuck this under something similar to what you have for `retentionPolicy` in the ClusterLogging api

We should have a larger discussion here with respect to who is defining what?  I understand the desire to hide this behind API but the general logic from the beginning has been: CLO provides abstract, EO allows more find tuning.  Given there is an opportunity to apply these performance improvents to any ES deployment we likely do not want to have CLO and Jaeger have to specify the same thing.  We do however, at a minimum need a way for the parent operator to tell the child to which indices we apply these changes.  This will allow a more flexible us of Elasticsearch by cluster logging and jaeger which is where we desire to go.  

I'm open to suggestions but I don't think there is anyway for EO and CLO to share the same retention policy API

Reply:
Simplified the api with the assumption the EO will infer the templates and indices to create